test/e2e_node/pod_host_ips.go

```diff
@@ -214,23 +172,34 @@ var _ = common.SIGDescribe("Dual Stack Host IP [Feature:PodHostIPs]", func() {
 			framework.ExpectNoError(err)
 			for _, node := range nodeList.Items {
 				if node.Name == p.Spec.NodeName {
-					nodeIPs := []string{}
+					got := sets.New[string]()
+					for _, hostIP := range p.Status.HostIPs {
+						got.Insert(hostIP.IP)
+					}
+
+					want := sets.New[string]()
 					for _, address := range node.Status.Addresses {
 						if address.Type == v1.NodeInternalIP {
-							nodeIPs = append(nodeIPs, address.Address)
+							want.Insert(address.Address)
 						}
 					}
-					gomega.Expect(p.Status.HostIPs).Should(gomega.Equal(nodeIPs))
+					if !got.Equal(want) {
+						framework.Failf("got %v, want %v", got, want)
+					}
 					break
 				}
 			}
 
 			ginkgo.By("deleting the pod")
-			err = podClient.Delete(ctx, pod.Name, *metav1.NewDeleteOptions(30))
+			err = podClient.Delete(ctx, pod.Name, *metav1.NewDeleteOptions(1))
 			framework.ExpectNoError(err, "failed to delete pod")
 		})
 
 		ginkgo.It("should provide hostIPs as an env var", func(ctx context.Context) {
+			if !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.PodHostIPs) {
+				e2eskipper.Skipf("PodHostIPs feature is not enabled")
```

Comment:
at least in e2e this does not work as people think, @pacoxu is this valid in this case?

Reply:
yeah, but `!utilfeature.DefaultFeatureGate.Enabled(kubefeatures.PodHostIPs) ` this only works on unit or integration tests that build the test and the code that exercise the test, in e2e you build an external binary that runs api calls against the code you want to test, so the featuregate utils here are not related, at least for the e2e that is not node, and I can't understand how this can work here

Reply:
https://github.com/kubernetes/kubernetes/blob/619be9c15372238edbb442e6b2ffd7a6cc827cfc/test/e2e_node/swap_test.go#L145-L148

https://github.com/kubernetes/kubernetes/blob/dd9c3358f5e99d4eb4f8a2d7dc77f59e896249e0/test/e2e_node/util.go#L526-L532

There are two other places in e2e_node that also use this

Reply:
yeahm, but I don't know if those tests are correct either, it will be easy to verify by running the test manually,

Reply:
Probably, we should add TODO item or track it somewhere.